lib/yarp.rb

```diff
@@ -102,6 +102,11 @@ def ==(other)
         other.end_offset == end_offset
     end
 
+    # Returns a new location that is the union of this location and the other.
+    def to(other)
+      Location.new(source, start_offset, other.end_offset - start_offset)
```

Comment:
What if other.start_offset < start_offset? Then this would return an incorrect result.
The logic should be similar to https://github.com/oracle/truffleruby/blob/4468b51ad29f7b4b197024e2baf9b0137eb75eed/src/main/java/org/truffleruby/parser/ast/ParseNode.java#L91-L98
So min() of start offsets and length = max() of end offsets - the result of min()

Reply:
I've added tests for that now.